streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -1209,14 +1210,14 @@ void process(final ProcessorSupplier<? super K, ? super V> processorSupplier,
      * internally generated name, and "-changelog" is a fixed suffix.
      * You can retrieve all generated internal topic names via {@link KafkaStreams#toString()}.
      *
-     * @param otherStream     the {@code KStream} to be joined with this stream
+     * @param otherStream     the {@code KStream}  (side input) to be joined with this (main input) stream
      * @param joiner          a {@link ValueJoiner} that computes the join result for a pair of matching records
      * @param windows         the specification of the {@link JoinWindows}
      * @param keySerde        key serdes for materializing the other stream,
```

Comment:
I am just realizing, that `other stream` is not right here either -- it's used for both input streams, as both will have the same key type.